networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/rpc/RpcHandler.java

```diff
@@ -157,10 +174,18 @@ public RpcStream getRpcStream() {
 
     @Override
     protected void channelRead0(final ChannelHandlerContext ctx, final ByteBuf msg) {
-      if (rpcRequestHandler != null) {
-        rpcRequestHandler.onData(nodeId, rpcStream, msg);
-      } else {
-        bufferedData.add(msg);
+      if (outputStreamClosed) {
+        // Discard any data if output stream has been closed
+        return;
+      }
+      try {
+        // TODO - we may want to optimize this to pass on ByteBuf's directly and manage their
+        //  garbage collection rather than immediately copying these bytes
+        final Bytes bytes = Bytes.wrapByteBuf(msg);
+        outputStream.write(bytes.toArray());
```

Comment:
We might want to have some kind of custom `InputStream` that runs over appended `ByteBuf` instances, similar to how we were previously managing these `ByteBuf`'s in `MessageBuffer`.  If we think that's worth doing, I can add a follow-up ticket.